phoenix-core/src/it/java/org/apache/phoenix/end2end/index/TransformIT.java

```diff
@@ -163,6 +165,183 @@ public void testSystemTransformTablePopulatedForTable() throws Exception {
+            assertMetadata(conn, PTable.ImmutableStorageScheme.SINGLE_CELL_ARRAY_WITH_OFFSETS, PTable.QualifierEncodingScheme.TWO_BYTE_QUALIFIERS, record.getNewPhysicalTableName());
+
+            String upsertStmt = "UPSERT INTO " + fullTableName + " (PK1, INT_PK, V1, V2) VALUES ('%s', %d, '%s', %d)";
+            // Now these live mutations should go into the index tables and the transforming table
+            conn.createStatement().execute(String.format(upsertStmt, "b", 2, "val2", 2));
+            conn.createStatement().execute(String.format(upsertStmt, "c", 3, "val3", 3));
+            assertEquals(2, getRowCount(conn, fullIdxName));
+            assertEquals(2, getRowCount(conn, record.getNewPhysicalTableName()));
+
+            // Delete one mutation
+            conn.createStatement().execute("DELETE FROM " + fullTableName + " WHERE PK1='b'");
+            assertEquals(1, getRowCount(conn, fullIdxName));
+            assertEquals(1, getRowCount(conn, record.getNewPhysicalTableName()));
+        }
+
+    }
+
+    @Test
+    public void testTransformForLiveMutations_mutatingBaseTableForView() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl(), testProps)) {
+            conn.setAutoCommit(true);
+            String schema = "S_" + generateUniqueName();
+            String tableName = "TBL_" + generateUniqueName();
+            String fullTableName = SchemaUtil.getTableName(schema, tableName);
+            String parentViewName = "VWP_" + generateUniqueName();
+            String viewName = "VW_" + generateUniqueName();
+            String viewIdxName = "VWIDX_" + generateUniqueName();
+
+            String createTableSql = "CREATE TABLE " + fullTableName + " (PK1 VARCHAR NOT NULL, INT_PK INTEGER NOT NULL, " +
+                    "V1 VARCHAR, V2 INTEGER CONSTRAINT NAME_PK PRIMARY KEY(PK1, INT_PK)) ";
+            conn.createStatement().execute(createTableSql);
+            assertMetadata(conn, ONE_CELL_PER_COLUMN, NON_ENCODED_QUALIFIERS, fullTableName);
+
+            String createParentViewSql = "CREATE VIEW " + parentViewName + " ( PARENT_VIEW_COL1 VARCHAR ) AS SELECT * FROM " + fullTableName;
+            conn.createStatement().execute(createParentViewSql);
+
+            String createViewSql = "CREATE VIEW " + viewName + " ( VIEW_COL1 INTEGER, VIEW_COL2 VARCHAR ) AS SELECT * FROM " + parentViewName;
+            conn.createStatement().execute(createViewSql);
+
+            String createViewIdxSql = "CREATE INDEX " + viewIdxName + " ON " + viewName + " (VIEW_COL1) include (VIEW_COL2) ";
+            conn.createStatement().execute(createViewIdxSql);
+
+            String upsertStmt = "UPSERT INTO " + viewName + " (PK1, INT_PK, V1, VIEW_COL1, VIEW_COL2) VALUES ('%s', %d, '%s', %d, '%s')";
+            conn.createStatement().execute(String.format(upsertStmt, "a", 1, "val1", 1, "col2_1"));
+
+            // Now do a transform and check still the index table is empty since we didn't build it
```

Comment:
comment's not accurate because we didn't build async and we check below that the table is populated